solutions/basic-quickstart/variables.tf

```diff
@@ -0,0 +1,226 @@
+  sensitive   = true
+  default     = null
+
+  validation {
+    condition     = (var.github_username == null && var.github_password == null) || (var.github_username != null && var.github_password != null)
+    error_message = "Either both 'github_password' and 'github_username' must be set, or neither."
+  }
+}
+
+variable "github_username" {
+  description = "A GitHub username is used to authenticate access to the private repository specified in the 'source_url' field."
+  type        = string
+  default     = null
+}
+
+##############################################################################
+# Code Engine App
+##############################################################################
+
+variable "app_name" {
+  description = "The name of the application to be created and managed. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<app_name>` format. [Learn more](https://cloud.ibm.com/docs/codeengine?topic=codeengine-application-workloads)"
+  type        = string
+  default     = "my-ce-app"
+}
+
+variable "app_scale_cpu_memory" {
+  description = "Define the amount of CPU and memory resources for each instance. [Learn more](https://cloud.ibm.com/docs/codeengine?topic=codeengine-mem-cpu-combo)"
+  type        = string
+  default     = "1 vCPU / 4 GB"
+}
+
+variable "app_image_port" {
+  description = "The port which is used to connect to the port that is exposed by the container image."
+  type        = number
+  default     = 8080
+}
+
+variable "managed_domain_mappings" {
+  description = "Define which of the following values for the system-managed domain mappings to set up for the application. [Learn more](https://cloud.ibm.com/docs/codeengine?topic=codeengine-application-workloads#optionsvisibility)"
+  type        = string
+  default     = "local_public"
+  validation {
+    condition     = var.managed_domain_mappings == null || can(regex("local_public|local_private|local", var.managed_domain_mappings))
+    error_message = "Valid values are 'local_public', 'local_private', or 'local'."
+  }
+}
+
+variable "app_scale_ephemeral_storage_limit" {
+  description = <<EOT
+The amount of ephemeral storage to set for the instance of the app.
+The units for specifying ephemeral storage are Megabyte (M) or Gigabyte (G), whereas G and M are the shorthand expressions for GB and MB. [Learn more](https://cloud.ibm.com/docs/codeengine?topic=codeengine-mem-cpu-combo#unit-measurements)
+
+
+The value must match regular expression '/^([0-9.]+)([eEinumkKMGTPB]*)$/'.
+EOT
+  type        = string
+  default     = "400M"
+}
+
+##############################################################################
+# Container Registry
+##############################################################################
+
+variable "container_registry_api_key" {
+  description = "The API key of the container registry in the target account. If this key is not provided, the IBM Cloud API key will be used instead."
```

Comment:
This should probably be more specific. This apikey will be used in the code engine secret that will be used for pulling the image that is built by the project right?